benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/services/internal/dao/ExplorationModelDAO.java

```diff
@@ -26,62 +30,182 @@ public ExplorationModelDAO(MongoClient mongoClient, BenchFlowTestModelDAO testMo
 
-    return benchFlowTestModel.getExplorationModel().getWorkloadUsersSpace();
+    return benchFlowTestModel.getExplorationModel().getGoalType();
+
   }
 
-  public synchronized void setWorkloadUserSpace(String testID, List<Integer> workloadUserSpace)
+  public synchronized void setGoalType(String testID, GoalType goalType)
       throws BenchFlowTestIDDoesNotExistException {
 
-    logger.info("setWorkloadUserSpace: " + testID);
+    logger.info("setGoalType: " + testID);
 
     final BenchFlowTestModel benchFlowTestModel = testModelDAO.getTestModel(testID);
 
-    benchFlowTestModel.getExplorationModel().setWorkloadUsersSpace(workloadUserSpace);
+    benchFlowTestModel.getExplorationModel().setGoalType(goalType);
 
     datastore.save(benchFlowTestModel);
+
+  }
+
+  public synchronized JavaCompatExplorationSpaceDimensions getExplorationSpaceDimensions(
+      String testID) throws BenchFlowTestIDDoesNotExistException {
+
+    logger.info("getExplorationSpaceDimensions: " + testID);
+
+    final BenchFlowTestModel benchFlowTestModel = testModelDAO.getTestModel(testID);
+
+    return benchFlowTestModel.getExplorationModel().getExplorationSpaceDimensions();
+
+  }
+
+  public synchronized void setExplorationSpaceDimensions(String testID,
+      JavaCompatExplorationSpaceDimensions explorationSpaceDimensions)
+      throws BenchFlowTestIDDoesNotExistException {
+
+    logger.info("setExplorationSpaceDimensions: " + testID);
+
+    final BenchFlowTestModel benchFlowTestModel = testModelDAO.getTestModel(testID);
+
+    benchFlowTestModel.getExplorationModel()
+        .setExplorationSpaceDimensions(explorationSpaceDimensions);
+
+    datastore.save(benchFlowTestModel);
+
   }
 
-  public synchronized ExperimentSelectionStrategy getExperimentSelectionStrategy(String testID)
+  public synchronized JavaCompatExplorationSpace getExplorationSpace(String testID)
       throws BenchFlowTestIDDoesNotExistException {
 
-    logger.info("getExperimentSelectionStrategy: " + testID);
+    logger.info("getExplorationSpace: " + testID);
 
     final BenchFlowTestModel benchFlowTestModel = testModelDAO.getTestModel(testID);
 
-    switch (benchFlowTestModel.getExplorationModel().getExperimentSelectionType()) {
-      case COMPLETE_SELECTION:
-        return new CompleteSelectionStrategy();
+    return benchFlowTestModel.getExplorationModel().getExplorationSpace();
+
+  }
+
+  public synchronized void setExplorationSpace(String testID,
+      JavaCompatExplorationSpace explorationSpace) throws BenchFlowTestIDDoesNotExistException {
+
+    logger.info("setExplorationSpace: " + testID);
+
+    final BenchFlowTestModel benchFlowTestModel = testModelDAO.getTestModel(testID);
+
+    benchFlowTestModel.getExplorationModel().setExplorationSpace(explorationSpace);
+
+    datastore.save(benchFlowTestModel);
+
+  }
+
+  public synchronized List<Integer> getExplorationPointIndices(String testID)
```

Comment:
Done in https://github.com/benchflow/benchflow/pull/394/commits/31ad2fa5fce3770738e2a0be4eeef66410c944d9